app/controllers/git_api/concept_exercises_controller.rb

```diff
@@ -61,5 +58,26 @@ def render_error(type)
         error: { type: type }
       }, status: 400
     end
+
+    private
+
+    def branch_name
+      @branch_name ||= begin
+        potential_branch_name = "#{params[:track_slug]}/#{params[:exercise_slug]}"
+        suffix = 1
+
+        while branch_name_not_unique(potential_branch_name)
+          potential_branch_name = "#{params[:track_slug]}/#{params[:exercise_slug]}-#{suffix}"
+          suffix += 1
+        end
+
+        potential_branch_name
+      end
+    end
+
+    def branch_name_not_unique(branch_name)
+      `git show-ref --verify --quiet refs/remotes/origin/#{branch_name}`
+      return $?.success?
```

Comment:
You can use [system](https://ruby-doc.org/core-2.7.0/Kernel.html#method-i-system) which returns true if the command is a success and false otherwise.

```suggestion
      system("git show-ref --verify --quiet refs/remotes/origin/#{branch_name}")
```